modules/builds-running-entitled-builds-with-sharedsecret-objects.adoc

```diff
@@ -0,0 +1,161 @@
+====
++
+[source,yaml]
+----
+$ oc apply -f -<<EOF
+apiVersion: sharedresource.openshift.io/v1alpha1
+kind: SharedSecret
+metadata:
+  name: etc-pki-entitlement
+spec:
+  secretRef:
+    name: etc-pki-entitlement
+    namespace: openshift-config-managed
+EOF
+----
+
+. Create a `ClusterRole` to grant permission to access the `SharedSecret` object:
++
+[source,yaml]
+----
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: use-share-etc-pki-entitlement
+rules:
+  - apiGroups:
+      - sharedresource.openshift.io
+    resources:
+      - sharedsecrets
+    resourceNames:
+      - etc-pki-entitlement
+    verbs:
+      - use
+----
+
+. Create a `Role` and `RoleBinding` object that grants the Shared Resources CSI driver the permission to access the `SharedSecret` object:
++
+[source,yaml]
+----
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: share-etc-pki-entitlement
+  namespace: openshift-config-managed
+rules:
+  - apiGroups:
+      - ""
+    resources:
+      - secrets
+    resourceNames:
+      - etc-pki-entitlement
+    verbs:
+      - get
+      - list
+      - watch
+----
++
+[source,yaml]
+----
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name: share-etc-pki-entitlement
+  namespace: openshift-config-managed
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: share-etc-pki-entitlement
+subjects:
+  - kind: ServiceAccount
+    name: csi-driver-shared-resource
+    namespace: openshift-builds <1>
+----
++
+<1> Enter the name of the namespace where openshift-builds is installed.
+
+. Create a `RoleBinding` object for the ServiceAccount that consumes the `SharedSecret` in your current namespace:
++
+[source,yaml]
+----
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name: use-share-etc-pki-entitlement
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: use-share-etc-pki-entitlement
+subjects:
+  - kind: ServiceAccount
+    name: pipeline
+  - kind: ServiceAccount
+    name: builder
+----
++
+[NOTE]
+====
+The ServiceAccount that consumes the `SharedSecret` are created and managed by the OpenShift controllers.
+====
+
+. Mount the `SharedSecret` with relevant mount details using the `buildah` buildStrategy:
++
+[source,yaml]
+----
+$ oc apply -f -<<EOF
+apiVersion: shipwright.io/v1beta1
+kind: Build
+metadata:
+  name: buildah-rhel
+spec:
+  source:
+    type: Git
+    git:
+      url: <LOCATION_OF_OPERATOR_ORGANIZATION>
+      revision: entitled-builds
+    contextDir: rh-builds
+  strategy:
+    name: buildah
+    kind: ClusterBuildStrategy
+  paramValues:
+  - name: dockerfile
+    value: ContainerFile
+  volumes:
+  - csi:
+      driver: csi.sharedresource.openshift.io
+      readOnly: true #<1>
+      volumeAttributes:
+        sharedSecret: etc-pki-entitlement #<2>
+    name: etc-pki-entitlement 
+  output:
+    image: <PATH_OF_INTERNAL_REGISTRY>
```

Comment:
End user/developer needs to provide this, doesn't have to be the internal registry. We have a footnote on this elsewhere in our docs: https://85265--ocpdocs-pr.netlify.app/openshift-builds/latest/work_with_builds/using-builds#ob-creating-a-buildah-build_using-builds

```suggestion
    image: <output-image-reference>
```